file/random_access_file_reader.cc

```diff
@@ -97,6 +97,15 @@ IOStatus RandomAccessFileReader::Read(
 
   IOStatus io_s;
   uint64_t elapsed = 0;
+  size_t alignment = file_->GetRequiredBufferAlignment();
+  bool is_aligned = false;
+  if (scratch != nullptr) {
+    // Check if offset, length and buffer are aligned.
+    is_aligned = (offset & (alignment - 1)) == 0 &&
+                 (n & (alignment - 1)) == 0 &&
+                 (uintptr_t(scratch) & (alignment - 1)) == 0;
```

Comment:
Is it the first (non-assertion) place that we assume alignment is a power of two?

Reply:
Oh, isn't alignment always the power of 2? I added this because that's how it is in posix. Is it possible it can be different on other file system?

Reply:
I don't know of any case where this assumption is invalid. It's fine with me

Reply:
It's always power of 2.

Reply:
Yes it is the first place.